cmd/zpool/zpool_vdev.c

```diff
@@ -1220,14 +1238,15 @@ make_disks(zpool_handle_t *zhp, nvlist_t *nv)
 		verify(!nvlist_lookup_uint64(nv, ZPOOL_CONFIG_WHOLE_DISK,
 		    &wholedisk));
 
-		if (!wholedisk) {
+		if (!wholedisk || part == ZPOOL_PARTITION_RAW) {
 			/*
 			 * Update device id string for mpath nodes (Linux only)
 			 */
-			if (is_mpath_whole_disk(path))
+			if (wholedisk || is_mpath_whole_disk(path))
 				update_vdev_config_dev_strs(nv);
 
-			(void) zero_label(path);
+			if (!is_spare(NULL, path))
```

Comment:
zero_label will fail to open the device because it's already open exclusively. It's already done for the legacy case below, so it should be done here as well.